pkg/providers/v1/aws_loadbalancer.go

```diff
@@ -618,8 +618,7 @@ func (c *Cloud) ensureTargetGroup(targetGroup *elbv2.TargetGroup, serviceName ty
 			return nil, fmt.Errorf("expected only one target group on CreateTargetGroup, got %d groups", len(result.TargetGroups))
 		}
 
-		tg := result.TargetGroups[0]
-		return tg, nil
+		targetGroup = result.TargetGroups[0]
```

Comment:
Seems like a simple fix. I wonder, @nckturner or @M00nF1sh, maybe this was coded this way originally because we wanted to sleep some period of time after creation of a TargetGroup before attempting to register expected instances in the TargetGroup below on original line 632?

Reply:
I encountered the same issue with 1.23.0 today where the target group had no target registered.
Restarting the controller seems to have correctly updated the target group.

The return might be there to make a sleep, but I am not sure we are doing the reconcile loop again unless there is a change in the service.

I will try your version and see if it's happening again.